datafusion/execution/src/memory_pool/proxy.rs

```diff
@@ -54,12 +60,15 @@ impl<T> VecAllocExt for Vec<T> {
     }
 }
 
-/// Extension trait for [`RawTable`] to account for allocations.
+/// Extension trait for hash browns [`RawTable`] to account for allocations.
 pub trait RawTableAllocExt {
     /// Item type.
     type T;
 
-    /// [Insert](RawTable::insert) new element into table and store additional allocated bytes in `accounting` (additive).
+    /// [Insert](RawTable::insert) new element into table and increase
+    /// `accounting` by any newly allocated bytes.
+    ///
+    /// Returns the bucket where the element was inserted.
```

Comment:
TBH this design is "good enough", but we should probably add some examples at some point (doesn't have to be in this PR though). Same is actually true for the other trait.

Reply:
Docs + remove custom strategy: https://github.com/apache/arrow-datafusion/pull/9058

docs:https://github.com/apache/arrow-datafusion/pull/9059